.github/workflows/builds.yml

```diff
@@ -0,0 +1,54 @@
+name: builds
+
+on:
+  release:
+    types: [created]
+  push:
+    branches:
+      - master
+      - main
+  pull_request:
+
+jobs:
+  docker-oss:
+    runs-on: ubuntu-latest
+    steps:
+      -
+        name: Checkout
+        uses: actions/checkout@v2
+      -
+        name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: |
+            name=rudderlabs/develop-rudder-server
+            name=rudderlabs/rudder-server,enable=${{ github.ref == format('refs/heads/{0}', 'master') || github.event_name == 'release' }}
+          tags: |
+            type=ref,event=branch
+            type=raw,value=latest,enable={{is_default_branch}}
+            type=raw,value=${{ github.head_ref }}
+            type=semver,pattern={{version}}
+            type=semver,pattern={{major}}.{{minor}}
+            type=semver,pattern={{major}}
+
+      -
+        name: Login to DockerHub
+        uses: docker/login-action@v2
+        with:
+          username: rudderlabs
```

Comment:
[minor] This could be a secret as well as showed in the first example [here](https://github.com/marketplace/actions/docker-login).